python/IECore/Enum.py

```diff
@@ -91,7 +93,9 @@ def __ne__( self, other ):
 
 		def __lt__( self, other ):
 
-			assert( type( self ) is type( other ) )
+			if type( self ) is not type( other ):
+				return False
```

Comment:
I'm less sure about this. Python 3's enum class would throw here, and this implementation leads to some surprising results whereby `a > b and b > a` is true :

```
>>> import IECore
>>> e1 = IECore.Enum.create( "A", "B" )
>>> e2 = IECore.Enum.create( "A", "B" )
>>> e1.A == e2.A
False
>>> e1.A < e2.A
False
>>> e1.A > e2.A
True
>>> e2.A < e1.A
False
>>> e2.A == e1.A
False
>>> e2.A > e1.A
True
```

Did you change this in response to a specific problem the assert was causing? If not, I suggest we consider matching the Python 3 behaviour and throw still.

Reply:
Note for later: I changed the behaviour of `__lt__` to be more in line with what Py3 Enum does. One key difference remains: `enum.Enum` odes not allow lesser or greater comparisons against another Enum instance, even if it's the same class.